### ISSUE TITLE
Make vlan available as a variable

### ISSUE THREAD

**Issue by dupondje** [ipxe/ipxe#1202]
When booting iPXE via UEFI, it automatically configures the UEFI PXE VLAN that is used, which is great.
But now it would be useful if the vlan was also available as a variable.

For example if you want to kickstart a CentOS install, the VM also needs its IP configured.
So you need to pass the vlan tag to the CentOS arguments, but the vlan is not available as a variable.

**Comment by mcb30** [comment]
Yes, this would be useful, and should be exposed via the netdevice settings mechanism in `netdev_settings.c`.

The only real question is what to call the variable.  `${vlan}` is potentially too generic a name and so likely to have been used in existing scripts.  There is also the question of whether we need to expose the 12-bit VLAN tag number (as returned by `vlan_tag()`), the whole 16-bit VLAN tag control information number (as returned by `vlan_tci()`), or both.  We could maybe have `${vlantag}` and `${vlantci}`?

**Comment by dupondje** [comment]
@mcb30 : Thanks for the very fast response :)

What about `${vlan-id}` or `${vlan-tag}` ? And I think only the tag is useful, don't see what additional info the tci will give.
But of course, it does not hurt to have `${vlan-tci}` ... So if both is same effort, I would just do both :)

**Comment by NiKiZe** [comment]
I don't see what tci gives, or how it could be used?
For name, maybe confusing but unambiguous `${netX/8021q-vid}`

**Comment by aleks-mariusz** [comment]
> The only real question is what to call the variable

is this REALLY blocked for two years now because we can't come up with what to call the variable? :-) i don't even know what tci is, i just need the vlan number so i don't have to hard code these values into my boot.ipxe's when generating the kernel command line params to pass installers

**Comment by mcb30** [comment]
> > The only real question is what to call the variable
> 
> is this REALLY blocked for two years now because we can't come up with what to call the variable? :-)

Honestly: it's not so much blocked as it is delayed until either (a) someone happens to suggest the perfect name that is unlikely to clash with anything, (b) someone happens to say "can I pay for this?", or (c) I personally need it as a feature.

**Comment by aleks-mariusz** [comment]
This is totally understandable.  We consumers of your project don't really have a whole lot of rights to demand features get implemented any sooner than you happen to have spare time found if there's no compensation.  I get where you're coming from and i appreciate how transparent you were with your reply.

As I'm a poor developer myself, I can't come in on choice B.  But what i _can_ offer is that i have **found a 'work-around'** (in a roundabout way) for the functionality i was looking for:
- set up your DHCP server for that VLAN to provide option 132, with the numeric value of your vlan.
- then in your ipxe script, you can use the following to pull that DHCP option's value out and use it dynamically in your scripts:

```
set vlan_id ${netX.dhcp/132:int16} # Use DHCP option 132 to determine VLAN we're on
```

This gets the netX (whichever was last device)'s DHCP option 132, and formats it as a 2-byte integer.   This will get me by (with no code change needed) for now, at least until such time if ever you decide to add this natively to ipxe.  Thanks for the amazing project just the same!
